src/extension.ts

```diff
@@ -185,22 +140,13 @@ export function activate(context: vscode.ExtensionContext) {
     : vscode.CompletionItem
   {
     let item = new vscode.CompletionItem(args.entry.new_prefix);
-    item.insertText = "";
     item.sortText = new Array(args.index + 2).join('0');
-    item.range = new vscode.Range(args.position, args.position);
-    let arg: CommandArgs = {
-      old_prefix: '',
-      new_prefix: '',
-      old_suffix: args.entry.old_suffix,
-      new_suffix: args.entry.new_suffix,
-    };
-    item.command = {
-      arguments: [arg],
-      command: COMMAND_NAME,
-      title: "accept completion",
-    };
-    item.insertText = args.entry.new_prefix;
-    item.range = new vscode.Range(args.position.translate(0, -args.old_prefix.length), args.position);
+    
+    item.insertText = new vscode.SnippetString(args.entry.new_prefix)
+      .appendTabstop()
+      .appendText(args.entry.new_suffix);
+
+    item.range = new vscode.Range(args.position.translate(0, -args.old_prefix.length), args.position.translate(0, args.entry.old_suffix.length));
```

Comment:
One thing to check here is that the text after the cursor is what we expect it to be. The most likely way for it to not be what we expect is if the user has multiple selections, and the text after the cursor is different in the multiple selections. Here's an example of a case where we should make sure not to replace the `..` after the second `foo`:
![image](https://user-images.githubusercontent.com/11807870/74917432-55472f00-537c-11ea-89e4-f88e564ddff2.png)

Reply:
hey Jacob, thanks for bringing this to my attention, this is actually the range property behaviour, it operates only on the main selection

![multiselct-cursor position](https://user-images.githubusercontent.com/60742964/75070826-bac22980-54fc-11ea-95c2-842426b02fb8.gif)